packages/orbit-components/src/Itinerary/ItineraryPart/index.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import * as React from "react";
+import styled from "styled-components";
+
+import { ItineraryPartProvider } from "./context";
+import Stack from "../../Stack";
+import ItineraryPartStatus from "./components/ItineraryPartStatus";
+import getSpacingToken from "../../common/getSpacingToken";
+import defaultTheme from "../../defaultTheme";
+
+import type { Props } from ".";
+
+const StyledWrapper = styled.div`
+  position: relative;
+  margin-bottom: ${getSpacingToken};
+`;
+
+StyledWrapper.defaultProps = {
+  theme: defaultTheme,
+};
+
+const ItineraryPart = ({ status, label, children, spaceAfter, dataTest }: Props) => {
+  const content = React.Children.toArray(children);
+  const refs = React.useRef([]);
+  const [calculatedWidth, setCalculatedWidth] = React.useState(0);
+
+  React.useEffect(() => {
+    if (refs.current) {
+      const widths = refs.current.map(r => r.clientWidth).filter(Number);
+      setCalculatedWidth(Math.max(...widths));
+    }
+  }, [setCalculatedWidth]);
+
+  const parts =
+    content &&
+    content.length > 0 &&
+    React.Children.map(content, (el, i) => {
+      return (
+        <ItineraryPartProvider
+          index={i}
+          last={i === content.length - 1}
+          isNextHidden={content[i + 1] && content[i + 1].props.hidden}
+          count={content.length}
+          isHidden={el.props.hidden}
+          hasStatus={!!status}
+          calculatedWidth={calculatedWidth}
+        >
+          {el.type.displayName === "ItineraryPartPlace"
+            ? // eslint-disable-next-line no-return-assign
+              React.cloneElement(el, { ref: r => (refs.current[i] = r) })
+            : el}
+        </ItineraryPartProvider>
+      );
+    });
+
+  return (
+    <StyledWrapper spaceAfter={spaceAfter} data-test={dataTest}>
+      {status ? (
```

Comment:
I would extract it to render function that would look like this, so we don't have to have duplicate of `<Stack direction="column">{parts}</Stack>`:

```jsx
const renderWithStatus = children => {
  if(status) {
    return (
      <ItineraryPartStatus type={status} label={label} offset={calculatedWidth}>
        {children}
      </ItineraryPartStatus>
  }
  return children;
}
return (
  <StyledWrapper spaceAfter={spaceAfter} data-test={dataTest}>
    {renderWithStatus(<Stack direction="column">{parts}</Stack>)}
  </StyledWrapper>
)
```

Reply:
There is a better way then, just to wrap parts in `Stack` and no need to create function then :)

Reply:
If there is, let's go with it. But I would generally avoid code duplication that can potentially result in mistakes when updating the component 🙂